website/docs/r/cognito_user_pool.markdown

```diff
@@ -89,6 +89,8 @@ The following arguments are supported:
 
 #### Schema Attributes
 
+  ~> **Note on attribute_data_type:** when defining an attribute date type of `String` or `Number` the respective attribute constraints block (e.g `string_attribute_constraints` or `number_attribute_contraints`) is required to prevent recreation of the defined schema(s). This requirement is true for both standard (e.g name, email) and custom schema attributes.
```

Comment:
😎 Minor fixes:

```suggestion
~> **NOTE:** When defining an `attribute_data_type` of `String` or `Number`, the respective attribute constraints configuration block (e.g `string_attribute_constraints` or `number_attribute_contraints`) is required to prevent recreation the Terraform resource. This requirement is true for both standard (e.g. name, email) and custom schema attributes.
```